client/layout/controller.js

```diff
@@ -64,11 +64,35 @@ const getPages = () => {
 };
 
 class Controller extends Component {
+	componentDidUpdate( prevProps ) {
+		const prevQuery = this.getQuery( prevProps.location.search );
+		const prevBaseQuery = this.getBaseQuery( prevProps.location.search );
+		const baseQuery = this.getBaseQuery( this.props.location.search );
+
+		if ( prevQuery.page > 1 && ! isEqual( prevBaseQuery, baseQuery ) ) {
+			updateQueryString( { page: 1 } );
```

Comment:
`updateQueryString` relies on `history.push()`, which adds a new element to the user's browser history. I think in this case it might make more sense to use `history.replace()`.

```suggestion
			history.replace( getNewPath( { page: 1 } ) );
```

You can import `history` and `getNewPath` from `@woocommerce/navigation` as well.

Reply:
Definitely makes more sense- thanks for that!